BitwardenShared/UI/Vault/Vault/VaultList/VaultListView.swift

```diff
@@ -360,6 +353,27 @@ struct VaultListView: View {
         .onDisappear {
             store.send(.disappeared)
         }
+        .apply { view in
+            if #available(iOS 16.0, *) {
+                view.modifier(
+                    ReviewModifier(
+                        isEligible: store.state.isEligibleForAppReview,
+                        afterClosure: {
+                            store.send(.appReviewPromptShown)
+                        }
+                    )
+                )
+            } else {
+                view.modifier(
+                    RequestReviewLegacyModifier(
+                        isEligible: store.state.isEligibleForAppReview,
+                        afterClosure: {
+                            store.send(.appReviewPromptShown)
+                        }
+                    )
+                )
+            }
+        }
```

Comment:
🎨 There's already a lot in this view, so it'd be nice if we could extract this to the modifier file. That also keeps the logic for the different versions together.

```suggestion
        .requestReview(isEligible: store.state.isEligibleForAppReview) {
            store.send(.appReviewPromptShown)
        }
```

```
extension View {
    func requestReview(isEligible: Bool, afterClosure: @escaping () -> Void) -> some View {
        apply { view in
            if #available(iOS 16.0, *) {
                view.modifier(
                    ReviewModifier(
                        isEligible: isEligible,
                        afterClosure: afterClosure
                    )
                )
            } else {
                view.modifier(
                    RequestReviewLegacyModifier(
                        isEligible: isEligible,
                        afterClosure: afterClosure
                    )
                )
            }
        }
    }
}
```